File: 1_50/1_two_sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        
        for (int i=0; i<nums.length; i++){
            int comp = target - nums[i];
          
            if (!map.containsKey(comp)){
                map.put(nums[i], i);
            } else {
                return new int[]{map.get(comp), i};
            }
        }
        return null;
    }
}

File: 1_50/1_two_sum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        num = dict()
        for idx, val in enumerate(nums):
            comp = target - val
            if comp not in num:
                # not yet seen the complement
                num[val] = idx
            else:
                return [num[comp], idx]
        return None
    
